0.9.2 fails to build with gcc 16 on macOS

```
:info:build [ 50%] Building CXX object src/libunicode/CMakeFiles/unicode.dir/convert.cpp.o
:info:build cd /opt/local/var/macports/build/libunicode-e319c1db/work/build/src/libunicode && /opt/local/bin/g++-mp-16 -DLIBUNICODE_USE_STD_SIMD -D_GLIBCXX_DEBUG -Dunicode_EXPORTS -I/opt/local/var/macports/build/libunicode-e319c1db/work/libunicode-0.9.2/src -I/opt/local/var/macports/build/libunicode-e319c1db/work/libunicode-0.9.2/src/libunicode/.. -pipe -Os -DNDEBUG -I/opt/local/include -D_GLIBCXX_USE_CXX11_ABI=0 -std=c++20 -arch ppc -mmacosx-version-min=10.6 -fPIC -fdiagnostics-color=always -Wall -Wextra -fdiagnostics-color=always -Wconversion -Wduplicated-cond -Wextra-semi -Wimplicit-fallthrough -Wlogical-op -Wmissing-declarations -Wno-unknown-pragmas -Wnull-dereference -Wpessimizing-move -Wredundant-move -Wsign-conversion -pedantic -MD -MT src/libunicode/CMakeFiles/unicode.dir/convert.cpp.o -MF CMakeFiles/unicode.dir/convert.cpp.o.d -o CMakeFiles/unicode.dir/convert.cpp.o -c /opt/local/var/macports/build/libunicode-e319c1db/work/libunicode-0.9.2/src/libunicode/convert.cpp
:info:build [ 53%] Building CXX object src/libunicode/CMakeFiles/unicode.dir/emoji_segmenter.cpp.o
:info:build cd /opt/local/var/macports/build/libunicode-e319c1db/work/build/src/libunicode && /opt/local/bin/g++-mp-16 -DLIBUNICODE_USE_STD_SIMD -D_GLIBCXX_DEBUG -Dunicode_EXPORTS -I/opt/local/var/macports/build/libunicode-e319c1db/work/libunicode-0.9.2/src -I/opt/local/var/macports/build/libunicode-e319c1db/work/libunicode-0.9.2/src/libunicode/.. -pipe -Os -DNDEBUG -I/opt/local/include -D_GLIBCXX_USE_CXX11_ABI=0 -std=c++20 -arch ppc -mmacosx-version-min=10.6 -fPIC -fdiagnostics-color=always -Wall -Wextra -fdiagnostics-color=always -Wconversion -Wduplicated-cond -Wextra-semi -Wimplicit-fallthrough -Wlogical-op -Wmissing-declarations -Wno-unknown-pragmas -Wnull-dereference -Wpessimizing-move -Wredundant-move -Wsign-conversion -pedantic -MD -MT src/libunicode/CMakeFiles/unicode.dir/emoji_segmenter.cpp.o -MF CMakeFiles/unicode.dir/emoji_segmenter.cpp.o.d -o CMakeFiles/unicode.dir/emoji_segmenter.cpp.o -c /opt/local/var/macports/build/libunicode-e319c1db/work/libunicode-0.9.2/src/libunicode/emoji_segmenter.cpp
:info:build In file included from [01m[K/opt/local/var/macports/build/libunicode-e319c1db/work/libunicode-0.9.2/src/libunicode/convert.cpp:3[m[K:
:info:build [01m[K/opt/local/var/macports/build/libunicode-e319c1db/work/libunicode-0.9.2/src/libunicode/convert_simd_impl.h:[m[K In function '[01m[Ksize_t unicode::detail::[01;32m[Kconvert_utf8_to_utf32_simd[m[K(const char*, size_t, char32_t*)[m[K':
:info:build [01m[K/opt/local/var/macports/build/libunicode-e319c1db/work/libunicode-0.9.2/src/libunicode/convert_simd_impl.h:49:36:[m[K [01;31m[Kerror: [m[K'[01m[Kfixed_size_simd[m[K' is not a member of '[01m[Kconvert_stdx[m[K' [[01;31m[K-Wtemplate-body[m[K]
:info:build    49 |         auto batch = convert_stdx::[01;31m[Kfixed_size_simd[m[K<signed char, simd_size> {};
:info:build       |                                    [01;31m[K^~~~~~~~~~~~~~~[m[K
:info:build [01m[K/opt/local/var/macports/build/libunicode-e319c1db/work/libunicode-0.9.2/src/libunicode/convert_simd_impl.h:49:52:[m[K [01;31m[Kerror: [m[Kexpected primary-expression before '[01m[Ksigned[m[K' [[01;31m[K-Wtemplate-body[m[K]
:info:build    49 |         auto batch = convert_stdx::fixed_size_simd<[01;31m[Ksigned[m[K char, simd_size> {};
:info:build       |                                                    [01;31m[K^~~~~~[m[K
:info:build [01m[K/opt/local/var/macports/build/libunicode-e319c1db/work/libunicode-0.9.2/src/libunicode/convert_simd_impl.h:50:82:[m[K [01;31m[Kerror: [m[K'[01m[Kelement_aligned[m[K' is not a member of '[01m[Kconvert_stdx[m[K' [[01;31m[K-Wtemplate-body[m[K]
:info:build    50 |         batch.copy_from(reinterpret_cast<signed char const*>(src), convert_stdx::[01;31m[Kelement_aligned[m[K);
:info:build       |                                                                                  [01;31m[K^~~~~~~~~~~~~~~[m[K
:info:build [01m[K/opt/local/var/macports/build/libunicode-e319c1db/work/libunicode-0.9.2/src/libunicode/convert_simd_impl.h:[m[K In function '[01m[Ksize_t unicode::detail::[01;32m[Kconvert_utf8_to_utf16_simd[m[K(const char*, size_t, char16_t*)[m[K':
:info:build [01m[K/opt/local/var/macports/build/libunicode-e319c1db/work/libunicode-0.9.2/src/libunicode/convert_simd_impl.h:158:36:[m[K [01;31m[Kerror: [m[K'[01m[Kfixed_size_simd[m[K' is not a member of '[01m[Kconvert_stdx[m[K' [[01;31m[K-Wtemplate-body[m[K]
:info:build   158 |         auto batch = convert_stdx::[01;31m[Kfixed_size_simd[m[K<signed char, simd_size> {};
:info:build       |                                    [01;31m[K^~~~~~~~~~~~~~~[m[K
:info:build [01m[K/opt/local/var/macports/build/libunicode-e319c1db/work/libunicode-0.9.2/src/libunicode/convert_simd_impl.h:158:52:[m[K [01;31m[Kerror: [m[Kexpected primary-expression before '[01m[Ksigned[m[K' [[01;31m[K-Wtemplate-body[m[K]
:info:build   158 |         auto batch = convert_stdx::fixed_size_simd<[01;31m[Ksigned[m[K char, simd_size> {};
:info:build       |                                                    [01;31m[K^~~~~~[m[K
:info:build [01m[K/opt/local/var/macports/build/libunicode-e319c1db/work/libunicode-0.9.2/src/libunicode/convert_simd_impl.h:159:82:[m[K [01;31m[Kerror: [m[K'[01m[Kelement_aligned[m[K' is not a member of '[01m[Kconvert_stdx[m[K' [[01;31m[K-Wtemplate-body[m[K]
:info:build   159 |         batch.copy_from(reinterpret_cast<signed char const*>(src), convert_stdx::[01;31m[Kelement_aligned[m[K);
:info:build       |                                                                                  [01;31m[K^~~~~~~~~~~~~~~[m[K
:info:build make[2]: *** [src/libunicode/CMakeFiles/unicode.dir/convert.cpp.o] Error 1
:info:build make[2]: *** Waiting for unfinished jobs....
:info:build make[2]: Leaving directory `/opt/local/var/macports/build/libunicode-e319c1db/work/build'
:info:build make[1]: *** [src/libunicode/CMakeFiles/unicode.dir/all] Error 2
:info:build make[1]: Leaving directory `/opt/local/var/macports/build/libunicode-e319c1db/work/build'
:info:build make: *** [all] Error 2
```

--- a/src/libunicode/convert_simd_impl.h
+++ b/src/libunicode/convert_simd_impl.h
@@ -8,7 +8,7 @@
 #include <cstdint>
 
 // clang-format off
-#if __has_include(<experimental/simd>) && defined(LIBUNICODE_USE_STD_SIMD) && !defined(__APPLE__) && !defined(__FreeBSD__)
+#if __has_include(<experimental/simd>) && defined(LIBUNICODE_USE_STD_SIMD) && !defined(_LIBCPP_VERSION)
     #define USE_STD_SIMD_CONVERT
     #include <experimental/simd>
     namespace convert_stdx = std::experimental;

--- a/src/libunicode/scan_simd_impl.h
+++ b/src/libunicode/scan_simd_impl.h
@@ -6,7 +6,7 @@
 #include <string_view>
 
 // clang-format off
-#if __has_include(<experimental/simd>) && defined(LIBUNICODE_USE_STD_SIMD) && !defined(__APPLE__) && !defined(__FreeBSD__)
+#if __has_include(<experimental/simd>) && defined(LIBUNICODE_USE_STD_SIMD) && !defined(_LIBCPP_VERSION)
     #define USE_STD_SIMD
     #include <experimental/simd>
     namespace stdx = std::experimental;

